cmd/snap-image/main.go

```diff
@@ -0,0 +1,153 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/jessevdk/go-flags"
+
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/logger"
+)
+
+var (
+	// Standard streams, redirected for testing.
+	Stdin  io.Reader = os.Stdin
+	Stdout io.Writer = os.Stdout
+	Stderr io.Writer = os.Stderr
+	// set to logger.Panicf in testing
+	noticef = logger.Noticef
+
+	errorPrefix = i18n.G("error: %v\n")
+
+	commands []*cmdInfo
+)
+
+// cmdInfo holds information needed to call parser.AddCommand(...).
+type cmdInfo struct {
+	name, shortHelp, longHelp string
+	builder                   func() flags.Commander
+	optDescs                  map[string]string
+	argDescs                  []argDesc
+	extra                     func(*flags.Command)
+}
+
+type argDesc struct {
+	name string
+	desc string
+}
+
+func addCommand(name string, builder func() flags.Commander, optDescs map[string]string, argDescs []argDesc) *cmdInfo {
+	info := &cmdInfo{
+		name:     name,
+		builder:  builder,
+		optDescs: optDescs,
+		argDescs: argDescs,
+	}
+	commands = append(commands, info)
+	return info
+}
+
+func parser() *flags.Parser {
+	flagopts := flags.Options(flags.PassDoubleDash | flags.HelpFlag)
```

Comment:
Just saying, this is probably duplicated from other commands, perhaps it should live "upstairs".